src/database/sqlite/mod.rs

```diff
@@ -490,6 +490,38 @@ impl SqliteConn {
         .expect("Db must not fail")
     }
 
+    /// Retrieves a limited and ordered list of transactions ids that happened during the given
+    /// range.
+    pub fn db_list_txids(&mut self, start: u32, end: u32, limit: u64) -> Vec<bitcoin::Txid> {
+        db_query(
+            &mut self.conn,
+            "SELECT DISTINCT(txid) FROM ( \
+                SELECT * from ( \
+                    SELECT txid, blocktime AS date FROM coins \
+                    WHERE blocktime >= (?1) \
+                    AND blocktime <= (?2) \
+                    ORDER BY blocktime DESC LIMIT (?3) \
+                ) \
+                UNION \
+                SELECT * FROM (
+                    SELECT spend_txid AS txid, spend_block_time AS date FROM coins \
+                    WHERE spend_block_time >= (?1) \
+                    AND spend_block_time <= (?2) \
+                    ORDER BY spend_block_time DESC LIMIT (?3) \
+                ) \
+                ORDER BY date DESC LIMIT (?3) \
```

Comment:
Is there a specific reason for having `DESC LIMIT (?3)` for inner queries? I think this is incorrect because you need to only limit the number of *distinct* txids.

For instance you have coins with deposit txids `[A, A, B, C, C, C]`, as well as coins with spend txids `[A, A, B, C]`. Now you make a query with a limit of `2`. The first subquery will give you  `[A, A]` and the second one `[A, A]`. After the `DISTINCT` in the outer query you will only have a single result: `[A]`.

I wonder whether the correct fix would be:
```suggestion
            "SELECT DISTINCT(txid) FROM ( \
                SELECT * from ( \
                    SELECT txid, blocktime AS date FROM coins \
                    WHERE blocktime >= (?1) \
                    AND blocktime <= (?2) \
                    ORDER BY blocktime \
                ) \
                UNION \
                SELECT * FROM (
                    SELECT spend_txid AS txid, spend_block_time AS date FROM coins \
                    WHERE spend_block_time >= (?1) \
                    AND spend_block_time <= (?2) \
                    ORDER BY spend_block_time \
                ) \
                ORDER BY date DESC LIMIT (?3) \
```

Or:
```suggestion
            "SELECT DISTINCT(txid) FROM ( \
                SELECT * from ( \
                    SELECT DISTINCT(txid), blocktime AS date FROM coins \
                    WHERE blocktime >= (?1) \
                    AND blocktime <= (?2) \
                    ORDER BY blocktime DESC LIMIT (?3) \
                ) \
                UNION \
                SELECT * FROM (
                    SELECT DISTINCT(spend_txid) AS txid, spend_block_time AS date FROM coins \
                    WHERE spend_block_time >= (?1) \
                    AND spend_block_time <= (?2) \
                    ORDER BY spend_block_time DESC LIMIT (?3) \
                ) \
                ORDER BY date DESC LIMIT (?3) \
```

Reply:
Actually i think the second solution would wreck the ordering. So let's go for the first one. I'll wait for your opinion though.